include/deal.II/lac/sparse_matrix.h

```diff
@@ -2377,6 +2384,35 @@ SparseMatrix<number>::end (const size_type r)
 }
 
 
+template <typename number>
+template <class StreamType>
+void SparseMatrix<number>::print_matrix_market(StreamType &out,
+					       const number threshold) const
+{
+  Assert(cols != 0, ExcNotInitialized());
+  Assert(val != 0, ExcNotInitialized());
+  Assert( threshold > 0, dealii::ExcMessage("Negative threshold!") ); 
+
+  //Print the header
+  out << "%%MatrixMarket matrix coordinate real general\n";
```

Comment:
Here, and in the other places, I think `'\n'` is appropriate. `std::endl` is `'\n'` plus a call to `std::flush`, which is relatively expensive because it transfers data from user space to the OS and probably to disk. That is not necessary for each line of output.